### PR TITLE
Added ubuntu-toolbox version 24.10, removed 23.10 (eol)

### PR DESCRIPTION
**Pull request by davidscholberg**
This PR adds an ubuntu-toolbox image for ubuntu 24.10 and removes version 23.10 as it has reached EOL.

I copied the files from 24.04 and made a few slight changes (obviously updated the version numbers but also had to explicitly install `unminimize` since it doesn't seem to be included with the image anymore). Here's everything that I changed from 24.04:

```diff
diff -urN images/ubuntu/24.04/Containerfile images/ubuntu/24.10/Containerfile
--- images/ubuntu/24.04/Containerfile   2024-10-16 21:41:33.928967339 -0400
+++ images/ubuntu/24.10/Containerfile   2024-10-16 22:09:08.875131964 -0400
@@ -1,8 +1,8 @@
-FROM docker.io/library/ubuntu:24.04
+FROM docker.io/library/ubuntu:24.10

 LABEL com.github.containers.toolbox="true" \
       name="ubuntu-toolbox" \
-      version="24.04" \
+      version="24.10" \
       usage="This image is meant to be used with the toolbox command" \
       summary="Base image for creating Ubuntu Toolbx containers" \
       maintainer="Ievgen Popovych <jmennius@gmail.com>"
@@ -21,9 +21,10 @@
 # Install ubuntu-minimal & ubuntu-standard
 # Install extra packages as well as libnss-myhostname
 COPY extra-packages /
-RUN sed -Ei '/apt-get (update|upgrade)/s/^/#/' /usr/local/sbin/unminimize && \
-    apt-get update && \
-    yes | /usr/local/sbin/unminimize && \
+RUN apt-get update && \
+    DEBIAN_FRONTEND=noninteractive apt-get -y install unminimize && \
+    sed -Ei '/apt-get (update|upgrade)/s/^/#/' /usr/bin/unminimize && \
+    yes | /usr/bin/unminimize && \
     DEBIAN_FRONTEND=noninteractive apt-get -y install \
         ubuntu-minimal ubuntu-standard \
         libnss-myhostname \
```

I'm not sure if any of the other steps in the Containerfile are out of date, but the image does work fine with toolbox from my testing of it.

### CODE REVIEW

**Comment by tom-reinders** [comment]
@davidscholberg Please update your PR following #1567.

This bug is in all `library/ubuntu` versions 24.04 and later.

**Review by tom-reinders — CHANGES_REQUESTED** [review]
Probably have to add `24.10` to https://github.com/containers/toolbox/blob/main/.github/workflows/ubuntu-images.yaml#L39

**Comment by davidscholberg** [comment]
@tom-reinders updated PR with 24.10 added to .github/workflows/ubuntu-images.yaml, and I also update the latest_release option in that file to 24.10.

**Review by tom-reinders — APPROVED** [review]

**Comment by Jmennius** [comment]
> @tom-reinders updated PR with 24.10 added to .github/workflows/ubuntu-images.yaml, and I also update the latest_release option in that file to 24.10.

So the latest_release should stay with the latest LTS version.

**Review by Jmennius — CHANGES_REQUESTED** [review]
And we also need to drop 23.10 as it is EOL.
This will also make the PR look like a rename of 23.10 files - easier to review :) (thanks for the explicit diff BTW)

**Comment by davidscholberg** [comment]
@Jmennius updated PR with revert of `latest_release` back to LTS, and removed 23.10 from that file, and deleted the 23.10 directory as well.

File: images/ubuntu/24.10/Containerfile [hunk 1]
```diff
@@ -11,19 +11,14 @@ LABEL com.github.containers.toolbox="true" \
 # Remove docker-gzip-indexes to help with "command-not-found"
 RUN rm /etc/apt/apt.conf.d/docker-gzip-indexes /etc/apt/apt.conf.d/docker-no-languages
 
-# Enable myhostname nss plugin for clean hostname resolution without patching
-# hosts (at least for sudo), add it right after 'files' entry. We expect that
-# this entry is not present yet. Do this early so that package postinst (which
-# adds it too late in the order) skips this step
-RUN sed -Ei 's/^(hosts:.*)(\<files\>)\s*(.*)/\1\2 myhostname \3/' /etc/nsswitch.conf
```
Comment (Jmennius) [inline images/ubuntu/24.10/Containerfile:18]: Removed this as `libnss-myhostname` now does the actual configuration for us.

**Review by Jmennius — APPROVED** [review]

**Comment by Jmennius** [comment]
@davidscholberg can you add a sign-off to the commit please? (`git commit --amend -s`, just be sure to fetch the updated branch)

**Comment by davidscholberg** [comment]
> @davidscholberg can you add a sign-off to the commit please? (`git commit --amend -s`, just be sure to fetch the updated branch)

Done.

**Pull request merged**